bpf/lib/encrypt.h

```diff
@@ -291,10 +281,10 @@ ipsec_maybe_redirect_to_encrypt(struct __ctx_buff *ctx, __be16 proto,
 		return CTX_ACT_OK;
 	}
 
-	if (!dst || !dst->flag_has_tunnel_ep)
+	if (!dst || !dst->flag_has_tunnel_ep || !dst->key)
```

Comment:
nit, can this go into the `ipsec_redirect_sec_id_ok` function, so policy is done all in one place?

Reply:
If we then also rename the function, sure ;). But tbh I actually like the idea of evaluating encryption-eligibility separately, once for source and once destination.

Reply:
ack. Let's see how well-aligned IPsec and Wireguard are by now, ideally they'd be using the same helpers ...